src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7569,7 +8049,7 @@ public static void Main()
     {
         if (true)
         {
-            object d1 = Dummy(new C(""a""), new C(""b"") is var x1, x1);
+            (object d1, object d2) = (Dummy(new C(""a""), (new C(""b"") is var x1), x1), x1);
```

Comment:
Added DeconstructionDeclarationStmt_03 and DeconstructionDeclarationStmt_04.

---
In reply to: [76288214](https://github.com/dotnet/roslyn/pull/13344#discussion_r76288214) [](ancestors = 76288214)